apps/builder/app/routes/logout.tsx

```diff
@@ -1,11 +1,49 @@
-import type { ActionFunctionArgs } from "@remix-run/server-runtime";
+import { json, type LoaderFunctionArgs } from "@remix-run/server-runtime";
+import createDebug from "debug";
 import { authenticator } from "~/services/auth.server";
-import { loginPath } from "~/shared/router-utils";
+import { builderAuthenticator } from "~/services/builder-auth.server";
+import {
+  getAuthorizationServerOrigin,
+  isBuilderUrl,
+} from "~/shared/router-utils/origins";
 
-export default function Logout() {
-  return null;
-}
+const debug = createDebug("Auth").extend("Logout");
 
-export const loader = async ({ request }: ActionFunctionArgs) => {
-  await authenticator.logout(request, { redirectTo: loginPath({}) });
+export const loader = async ({ request }: LoaderFunctionArgs) => {
+  debug("Logout request received", request.url);
+
+  try {
+    if (isBuilderUrl(request.url)) {
+      debug("Temporary redirect to /tmp/login after logout");
+      await builderAuthenticator.logout(request, {
+        redirectTo: `${getAuthorizationServerOrigin(request.url)}/tmp/login`,
+      });
+    } else {
+      await authenticator.logout(request, { redirectTo: "/" });
+    }
+  } catch (error) {
+    if (error instanceof Response) {
+      const contentType = request.headers.get("Content-Type");
+
+      if (contentType?.includes("application/json")) {
+        const headers = new Headers(error.headers);
+        headers.set("Content-Type", "application/json");
+        return new Response(
+          JSON.stringify({
+            success: true,
+          }),
+          {
+            headers,
+            status: 200,
+          }
+        );
+      }
+
+      return error;
+    }
+
+    throw error;
+  }
+
+  return json({});
```

Comment:
shouldn't this contain at least something? like the success state or so

Reply:
changed on `throw new Error("Should not reach this point");`